library/cpp/include/wavemap/pipeline/pipeline.h

```diff
@@ -26,43 +29,62 @@ class Pipeline {
         thread_pool_(thread_pool ? std::move(thread_pool)
                                  : std::make_shared<ThreadPool>()) {}
 
+  // Copy construction is not supported
+  Pipeline(const Pipeline&) = delete;
+
+  //! Deregister all measurement integrators and map operations
   void clear();
 
+  //! Returns true if an integrator with the given name has been registered
   bool hasIntegrator(const std::string& integrator_name) const;
+  //! Deregister the integrator with the given name. Returns true if it existed.
   bool eraseIntegrator(const std::string& integrator_name);
+  //! Create and register a new integrator
   IntegratorBase* addIntegrator(const std::string& integrator_name,
                                 const param::Value& integrator_params);
+  //! Register the given integrator, transferring ownership
   IntegratorBase* addIntegrator(const std::string& integrator_name,
                                 std::unique_ptr<IntegratorBase> integrator);
+  //! Get a pointer to the given integrator, returns nullptr if it does not
+  //! exist
   IntegratorBase* getIntegrator(const std::string& integrator_name);
+  //! Access all registered integrators (read-only)
   const IntegratorMap& getIntegrators() { return integrators_; }
+  //! Deregister all integrators
   void clearIntegrators() { integrators_.clear(); }
 
+  //! Create and register a new map operation
   MapOperationBase* addOperation(const param::Value& operation_params);
+  //! Register the given map operation, transferring ownership
   MapOperationBase* addOperation(std::unique_ptr<MapOperationBase> operation);
+  //! Access all registered map operations (read-only)
   const OperationsArray& getOperations() { return operations_; }
+  //! Deregister all map operations
   void clearOperations() { operations_.clear(); }
 
+  //! Integrate a given measurement
   template <typename MeasurementT>
   bool runIntegrators(const std::vector<std::string>& integrator_names,
                       const MeasurementT& measurement);
+  //! Run the map operations
   void runOperations(bool force_run_all = false);
```

Comment:
Should this return a `bool` to stay consistent with `runIntegrators` and `runPipeline`?

Reply:
`runIntegrators` returns `false` if one name in `integrator_names` does not exist (which is a user error), and `runPipeline`, which internally calls `runIntegrators` and `runOperations`, forwards this error. But running the operations always succeeds, so I don't think it needs a return value.

Reply:
Ok makes sense.